content/docs/rfcs/72/README.md

```diff
@@ -0,0 +1,297 @@
+	version: "string",
+	credentials: {
+    		"membershipHash": {
+      			crypto: {
+        			cipher: "string",
+        			cipherparams: {
+					iv: "string",
+				},
+        			ciphertext: "string",
+        			kdf: "string",
+        			kdfparams: {
+          				dklen: integer,
+          				c: integer,
+          				prf: "string",
+          				salt: "string",
+        			},
+        			mac: "string",
+      			}
+    		}
+	}
+}
+
+```
+
+# Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, 
+“NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The keystore MUST be generated by a cryptographic construction with password verification and decryption.
+
+Keystore modules MUST include metadata, key derivation function, checksum, cipher, and a membership hash.
+
+## Metadata:
+Information about the keystore SHOULD be stored in the metadata.
+
+The declaration of `application`, `version`, and `appIdentifier` COULD occur in the metadata.
+
+- `application` : current application, MUST be a string
+
+- `version` : application version, MUST be a string
+
+- `appIdentifier`: application identifier, MUST be a string
+
+## Credentials:
+
+After RLN credentials are generated, it MUST be stored in a JSON schema.
+The Waku RLN credentials MUST consist of a `membershipHash` and `WakuCredential`.
+The `membershipHash` will be an identity hash of the user.
+The `WakuCredential` will store to encryption portion of the keystore.
+There COULD be multiple credentials stored in a keystore, categorized by the `membershipHash`.
+
+Each contruct MUST include the keypair:
+> key: [`membershipHash`]: pair: [`WakuCredential`]
+
+### membershipHash 
+
+The `membershipHash` SHOULD be generated by user's participating in a membership group.
+Each user SHOULD register to the group with an `identity_commitment` stored in a Merkle tree.
+A cryptographic hash function that SHOULD be used to generate the `membershipHash` is [SHA256](https://www.rfc-editor.org/rfc/rfc4634.txt), 
+other hash functions MAY be used.
+To generate the `membershipHash`, 
+the `treeIndex`, `membershipContract`, and `identityCredential` attributes SHOULD be used to create a hexadecimal string.
+- it MUST NOT already exist in the keystore.
+
+`treeIndex`
+
+After a user registers to a group, 
+a tree index value of the position in the Merkle tree SHOULD be returned.
+- it MUST be a Merkle tree data structure filled with `identity_commitment` from user registrations.
+- it MUST be a hexadecimal string
+
+`membershipContract`
+
+For decentralized membership registrations, 
+the `membershipContract` SHOULD be derived from a public blockchain using smart contracts.
+For centralized requirements, 
+the `membershipContract` SHOULD be derived from private servers. 
+- it MUST be a hash of a `contractId` and `contractAddress`
```

Comment:
the membership contract is just the address of the contract, not a hash of the contractId and contractAddress